src/backend/utils/adt/fssfuncs.c

```diff
@@ -0,0 +1,222 @@
+/*-------------------------------------------------------------------------
+ *
+ * fssfuncs.c
+ *		Filtered Space Saving functions
+ *
+ * Copyright (c) 2013-2015, PipelineDB
+ *
+ * IDENTIFICATION
+ *	  src/backend/utils/adt/fssfuncs.c
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+
+#include "access/htup_details.h"
+#include "catalog/pg_type.h"
+#include "lib/stringinfo.h"
+#include "libpq/pqformat.h"
+#include "nodes/nodeFuncs.h"
+#include "pipeline/fss.h"
+#include "utils/array.h"
+#include "utils/bytea.h"
+#include "utils/builtins.h"
+#include "utils/datum.h"
+#include "utils/fssfuncs.h"
+#include "utils/typcache.h"
+
+#define DEFAULT_K 5
+
+static FSS *
+fss_recv(struct varlena *bytes)
```

Comment:
This should probably be renamed because it's not a `*recv` function in the typical sense.